stew/byteutils.nim

```diff
@@ -46,11 +46,16 @@ func hexToByteArrayImpl(hexStr: openArray[char], output: var openArray[byte], fr
   ## Allows specifying the byte range to process into the array
   var sIdx = skip0xPrefix(hexStr)
 
+  if hexStr.len - sIdx < 1:
```

Comment:
this cannot happen simply because skip0xPrefix is already checking the length the same way - the code will never trigger - if you remove this change, your test case still correctly runs

Reply:
Hmm, in my case removing this does cause my tests in the `nim-web3` repo to fail 🤔 

Seems that it's my use of `openArray` rather than the fixed `byte` length which is causing the issue. I will add more tests for this.

Reply:
I also think this should be removed, per the arguments regarding empty output arrays in the discussion below.